SPLASH/src/ImGuiWindows/Chat/ChatWindow.cpp

```diff
@@ -0,0 +1,353 @@
+#include "pch.h"
+#include "ChatWindow.h"
+#include "imgui.h"
+#include "Sail/Application.h"
+#include "Network/NWrapperSingleton.h"
+
+#include "../SPLASH/src/game/events/TextInputEvent.h"
+#include "../SPLASH/src/game/events/NetworkChatEvent.h"
+#include "../SPLASH/src/game/events/NetworkJoinedEvent.h"
+#include "../SPLASH/src/game/events/NetworkDisconnectEvent.h"
+#include "Sail/events/types/NetworkPlayerChangedTeam.h"
+#include "Sail/events/types/WindowResizeEvent.h"
+
+ChatWindow::ChatWindow(bool showWindow) :
+	m_message(""),
+	m_timeSinceLastMessage(0.0f),
+	m_fadeTime(-1.0f),
+	m_fadeThreshold(-1.0f),
+	m_backgroundOpacityMul(0.4f),
+	m_scrollToBottom(false),
+	m_messageLimit(30),
+	m_messageSizeLimit(30),
+	m_retainFocus(true)
+{
+	m_app = Application::getInstance();
+	m_settings = &m_app->getSettings();
+	m_imguiHandler = m_app->getImGuiHandler();
+	m_eventDispatcher = &EventDispatcher::Instance();
```

Comment:
m_network is fetched here properly, but NWrapperSingleton::getInstance() is still used in 'renderChat'. Use m_eventDispatcher instead!